src/pages/dashboard/analysis/model.tsx

```diff
@@ -0,0 +1,75 @@
+import { AnyAction, Reducer } from 'redux';
+
+import { EffectsCommandMap } from 'dva';
+import { AnalysisData } from './data.d';
+import { fakeChartData } from './service';
+
+export type Effect = (
+  action: AnyAction,
+  effects: EffectsCommandMap & { select: <T>(func: (state: AnalysisData) => T) => T },
+) => void;
+
+export interface ModelType {
+  namespace: string;
+  state: AnalysisData;
+  effects: {
+    fetch: Effect;
+    fetchSalesData: Effect;
+  };
+  reducers: {
+    save: Reducer<AnalysisData>;
+    clear: Reducer<AnalysisData>;
+  };
+}
+
+const initState = {
+  visitData: [],
+  visitData2: [],
+  salesData: [],
+  searchData: [],
+  offlineData: [],
+  offlineChartData: [],
+  salesTypeData: [],
+  salesTypeDataOnline: [],
+  salesTypeDataOffline: [],
+  radarData: [],
+};
+
+const Model: ModelType = {
+  namespace: 'dashboardAndanalysis',
+
+  state: initState,
+
+  effects: {
+    *fetch(_, { call, put }) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [expected parameter: '{ call, put }' to have a typedef](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4583897)